### PR TITLE
Mime types were not set by default, so enable the mime factory

### PR DESCRIPTION
**Pull request by halkeye**
jenkins.io has tighter restrictions on mime types so i didn't notice it on early testing

Before:

image

After:

image

### CODE REVIEW

**Comment by martinda** [comment]
Thanks @halkeye . Is this feature testable? Could a test be written for it?

**Comment by halkeye** [comment]
I don't know. Java is not my strong suit. Is it possible to populate the public directory just for a test? Maybe somewhere in spring boot you could point public to a test directory?

**Comment by halkeye** [comment]
Okay I did a bunch of searching. I have no idea how to set that up. Is other at-configuration or MVC factory stuff tested?

**Comment by halkeye** [comment]
Reopening torerun

**Comment by sladyn98** [comment]
@martinda @halkeye One solution is get this merged in and once the controller tests are also merged in we can add check the mime type of the files, if we want to see if this is effective.

**Review by sladyn98 — APPROVED** [review]
@martinda @halkeye One solution is get this merged in and once the controller tests are also merged in we can add check the mime type of the files, if we want to see if this is effective.

File: src/main/java/com/org/jenkins/custom/jenkins/distribution/service/MimeMapping.java
Comment (sladyn98) [inline src/main/java/com/org/jenkins/custom/jenkins/distribution/service/MimeMapping.java:9]: Just a minor question should this be 
`public class ServletCustomizer implements WebServerFactoryCustomizer<TomcatServletWebServerFactory>` since we are using a TomCat server as our default

Comment (halkeye) [reply src/main/java/com/org/jenkins/custom/jenkins/distribution/service/MimeMapping.java:9]: Pretend I know nothing. Cause I know nothing.

This solution seems to work but I make no promises. As the expert I defer to you.

Comment (sladyn98) [reply src/main/java/com/org/jenkins/custom/jenkins/distribution/service/MimeMapping.java:9]: `TomcatServletWebServerFactory` is a subclass of `ConfigurableServletWebServerFactory` and would probably serve for more specific use cases, but this looks good enough. 
CC @martinda @kwhetstone

Comment (martinda) [reply src/main/java/com/org/jenkins/custom/jenkins/distribution/service/MimeMapping.java:9]: It is a net improvement, I added a issue to track testing later.

**Pull request merged**
